lib/experimental/fonts/font-library/class-wp-font-collection.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Font Collection class.
+ *
+ * This file contains the Font Collection class definition.
+ *
+ * @package    WordPress
+ * @subpackage Font Library
+ * @since      6.4.0
+ */
+
+if ( class_exists( 'WP_Font_Collection' ) ) {
+	return;
+}
+
+/**
+ * Font Collection class.
+ *
+ * @since 6.4.0
+ */
+class WP_Font_Collection {
+
+	/**
+	 * Font collection configuration.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @var array
+	 */
+	private $config;
+
+	/**
+	 * WP_Font_Collection constructor.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string $id     Font collection id.
+	 * @param array  $config Font collection config options.
+	 * @throws Exception If the required parameters are missing.
+	 */
+	public function __construct( $id, $config ) {
+
+		if ( empty( $id ) && is_string( $id ) ) {
```

Comment:
The `try/catch` eliminates the need to throw a fatal error here.

To eliminate the `try/catch, what if instead `WP_Error` is used and returned? That would eliminate the `try/catch` code.

Reply:
Could you provide an example of what you mean, please?

Reply:
There are a couple of ways to do this:

* Return `WP_Error()`.
* Create an InvalidArgument exception that returns `WP_Error()`.

I think for now the first option is more straightforward.

https://github.com/WordPress/gutenberg/pull/53816/commits/0e6c026c87933f64dae78049cc9bb53c7fa8dffb

Reply:
Fair points. If the class is instantiated outside of the Font Library's usage, then yes, unexpected results can happen.

Let me see how this handled within Core for some inspiration. I'll revert or modify the approach.

Reply:
Wait duh 🤦‍♀️ Oh silly me ... of course. The constructor returns that instance of its class. Imagine me flush faced 🤣 

Let's back up. What is the purpose of the `try/catch`? 

It's avoiding a fatal error and then recasting to an instance of `WP_Error`. 

Is that necessary? How are other endpoints handling fatal errors? Hmm, I'm curious. Time to explore other endpoints in Core.

Reply:
Another way to go .. thinking out loud ... is to very the `$id` and `$config` is correct _before_ instantiating `WP_Font_Collection`, i.e. check both in `WP_Font_Library::register_font_collection()`. By doing so, the `try/catch` gets removed and the validation happens where it's passed in for registration.

Let me do an experiment.

Reply:
Aha, inspiration ...

What's bothering me is why the `try/catch` is needed. In reviewing, it is not needed. Registering a font collection happens within the extender's plugin or theme as they will invoke `WP_Font_Library::register_font_collection()`. It does not go through REST. So a fatal error is appropriate and will provide the developer the information they need to debug their custom code.

In going further into the design, the current implementation uses a filter approach for extenders to register their font collections. This can be simplified by removing the filter and instead directly calling `WP_Font_Library::register_font_collection()`. Why? It'll be more performant (no hooks to call and process) while giving a more clear stack backtrace if the `$id` or `$config` arguments are invalid - with a fatal error thrown.

@matiasbenedetto what do you think?

Reply:
Thinking about this more ... the preferred approach in Core is for extenders to use a global function, rather than the static method call:

Rather than:
```
WP_Font_Library::register_font_collection( $id, $config );
```

instead extenders call:
```
wp_register_font_collection( $id, $config );
```

Reply:
Cool, based on you latest comments, this commit https://github.com/WordPress/gutenberg/pull/53816/commits/4e952e3360a42c97b09cbd21c183827130abdfec removes the  `wp_register_font_collection` filter and implements the function with the same name. Apart from that I removed the try catch from the WP_Font_Library::register_collection() method, so if the $config is wrong the fatal error will be shown.

Reply:
Apart from that, the replacement of the filter by the `wp_register_font_collection()` function allow us to use just one parameter `wp_register_font_collection( $config )` instead of the previous 2 `$id, $config`. I added a little refactor around that too in https://github.com/WordPress/gutenberg/commit/4e952e3360a42c97b09cbd21c183827130abdfec

Reply:
> function allow us to use just one parameter ($config) instead of the previous 2 ($id, $config)

Oh nice! 👏